src/node_state.rs

```diff
@@ -111,12 +111,20 @@ impl NodeState {
         Ok(())
     }
 
-    fn handle_new_block(&mut self, block: Block) -> Result<(), Error> {
+    fn handle_new_block(&mut self, mut block: Block) -> Result<(), Error> {
         self.clear_timeouts(&block.height);
         self.current_height = block.height;
         let txs_count = block.txs.len();
         let block_hash = block.hash();
 
+        block.txs.sort_by(
+            |tx1, tx2| match (&tx1.transaction_data, &tx2.transaction_data) {
+                (TransactionData::RegisterContract(_), _) => std::cmp::Ordering::Greater,
+                (_, TransactionData::RegisterContract(_)) => std::cmp::Ordering::Less,
+                (_, _) => std::cmp::Ordering::Equal,
+            },
+        );
```

Comment:
this is not related to PR, is it possible to have it in a different PR ?

Reply:
and what is the purpose of this ? why do we need the registering of contract first ? Do we relly consider we can have a transaction in the same block the contract is registered ? It's so unlikely (unless maybe in tests) and we pay the "cost" of sorting an array for that. I don't know if it is worthit